internal/errors/errors.go

```diff
@@ -0,0 +1,5 @@
+package errors
+
+import "errors"
+
+var ErrResourceNotFound = errors.New("resource not found")
```

Comment:
While this isn't actually a constant, it's being used like one. Maybe we should just move it to the `constants.go` file. It's also a bit weird (but that's a Go problem), to have a file named `error.go` that imports a package named in the same way